packages/navigator/src/browser/navigator-filter.ts

```diff
@@ -22,8 +22,6 @@ import { PreferenceChangeEvent } from '@theia/core/lib/browser/preferences';
 import { FileSystemPreferences, FileSystemConfiguration } from '@theia/filesystem/lib/browser/filesystem-preferences';
 import { FileNavigatorPreferences, FileNavigatorConfiguration } from './navigator-preferences';
 
-const FILES_EXCLUDE_PREFERENCE: keyof FileSystemConfiguration = 'files.exclude';
```

Comment:
fyi: not all strings in TS are just string, some can be used for property access, i.e `myObject['myProp']`
preference proxies are statically typed, one cannot use other prop names and type is properly determined for a prop, there is NO need for a constant. I had to remove it here because it confused compiler. The type of `FILES_EXCLUDE_PREFERENCE` now is `'files.exclude' | 'files.enableTrash'`, so compiler cannot derive any more  a proper type where `FILES_EXCLUDE_PREFERENCE` is used.

cc @svor @kittaakos